webpacker_guide.md

```diff
@@ -14,16 +14,75 @@ After reading this guide, you will know:
 
 ## What Is Webpacker?
 
+Webpacker is a Rails wrapper around the [webpack](https://webpack.js.org) build system that provides a standard webpack configuration and reasonable defaults. 
 
+### What is webpack?
+
+The goal of webpack, or any front-end build system, is to allow you to write your front-end code in a way that is convenient for developers and then package that code in a way that is convenient for browsers. With webpack, you can manage JavaScript, CSS, and static assets like files or fonts. Webpack will allow you to write your code, reference other code in your application, transform you code, and combine your code into easily downloadable packs.
 
 ## How is Webpacker Different from Sprockets?
 
+Rails also ships with Sprockets, an asset-packaging tool whose features overlap with Webpacker. Both tools will compile your JavaScript into into browser-friendly files, and minify and fingerprint them in production. Both tools allow you to incrementally change files in development.
+
+Sprockets, which was designed to be used with Rails, is somewhat simpler to integrate. In particular, code can be added to Sprockets via a Ruby gem. However, webpack is better at integrating with more current JavaScript tools and NPM packages, and allows for a wider range of integration.
+
+You should choose webpacker over Sprockets on a new project, if you want to use NPM packages, and if you want access to the most current JavaScript features and tools. You should choose Sprockets over Webpacker for legacy applications where migration might be costly, if you want to integrate using Gems, or if you have a very small amount of code to package. 
+
+If you are familiar with Sprockets, the following guide might give you some idea of how to translate. Please note that each tool has a slightly different structure, and the concepts don't directly map onto each other 
+
+|Task              | Sprockets         | Webpacker         |
+|------------------|-------------------|-------------------|
+|Attach JavaScript |javascript_link_tag|javascript_pack_tag|
+|Attach CSS        |stylesheet_link_tag|stylesheet_pack_tag|
+|Link to an image  |image_url          |image_pack_tag     |
+|Link to an asset  |asset_url          |asset_pack_tag     |
+|Require a script  |//= require        |require or include |
+
 ## Installing Webpacker
 
+In order to use Webpacker you must be using the Yarn package manager, version 1.x or up, and you must have Node.js installed, version 10.13.0 and up.
+
+Webpacker is installed by default in Rails 6.0 and up. In an older version, you can install it when a new project is created by adding `--webpack` to a `rails new` command. In an existing project, webpacker can be added by installing `bundle exec rails webpacker:install`. This installation command creates local files:
+
+* The Webpacker configuration at `config/webpacker.html`
+* Configuration files for Babel, PostCSS, and Browserslist
```

Comment:
`babel.config.js` and `postcss.config.js` in app root. I don't know where the browsers list file ends up. Is it worth a short sentence each describing what these files are for, or is that out of scope?

Reply:
We should probably explain each of the config files with a single sentence and link to further documentation.
You can find the Browserlist Configuration file in the app root `.browserslistrc`, but you need to display your hidden files, since its filename is prefixed with a dot.

Reply:
How much more description do we need other than "configuration file"?

Reply:
In this case I would just link to the technology docs, I can open a PR for that.
I think it's a general question for this guide.
Should this guide make it possible to understand the ecosystem that comes with Webpack/Webpacker or should it give a overview about Webpacker on condition that someone is familiar with Webpack and everything JS related?
What do you think?

Reply:
> In this case I would just link to the technology docs, I can open a PR for that.
> I think it's a general question for this guide.
> Should this guide make it possible to understand the ecosystem that comes with Webpack/Webpacker or should it give a overview about Webpacker on condition that someone is familiar with Webpack and everything JS related?
> What do you think?

I think this guide needs to enable somebody who's not familiar with Webpack to get things up and running. The nitty gritty details of customizing Webpack could go into the Webpacker gem documentation.